File: running-sum-of-1d-array.cpp
#include <vector>

class Solution {
public:
    std::vector<int> runningSum(std::vector<int>& nums) {
        // Start from the second element (index 1)
        for (int i = 1; i < nums.size(); i++) {
            // Add the previous element's value to the current element
            nums[i] += nums[i - 1];
        }
        return nums;
    }
};

File: Running Sum of 1d Array.py
from typing import List

class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        # Start iterating from the second element (index 1)
        for i in range(1, len(nums)):
            # Add the value of the previous element to the current one
            nums[i] += nums[i - 1]
        
        return nums